slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala

```diff
@@ -164,12 +173,185 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
+          */
+        java.sql.Types.VARCHAR
+      }
+      override def setValue(v: LocalTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, v.toString)
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalTime = {
+        LocalTime.parse(r.getString(idx))
+      }
+      override def updateValue(v: LocalTime, r: ResultSet, idx: Int) = r.updateString(idx, v.toString)
+
+      override def valueToSQLLiteral(value: LocalTime) = {
+        value match {
+          case null => "NULL"
+          case _ => s"'${value.toString}'"
+        }
+      }
+    }
+
+    class LocalDateTimeJdbcType extends DriverJdbcType[LocalDateTime] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = java.sql.Types.TIMESTAMP
+      override def setValue(v: LocalDateTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setTimestamp(idx, Timestamp.valueOf(v))
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalDateTime = {
+        r.getTimestamp(idx).toLocalDateTime
+      }
+      override def updateValue(v: LocalDateTime, r: ResultSet, idx: Int) : Unit = {
+        r.updateTimestamp(idx, Timestamp.valueOf(v))
+      }
+      override def hasLiteralForm = true
+      override def valueToSQLLiteral(value: LocalDateTime) = {
+        value match {
+          case null => "NULL"
+          case _ =>  s"'${Timestamp.valueOf(value)}'"
+        }
+      }
+    }
+
+    class InstantJdbcType extends DriverJdbcType[Instant] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = {
+        java.sql.Types.TIMESTAMP
+      }
+      override def setValue(v: Instant, p: PreparedStatement, idx: Int) : Unit = {
+        p.setTimestamp(idx, Timestamp.from(v))
+      }
+      override def getValue(r: ResultSet, idx: Int) : Instant = {
+        r.getTimestamp(idx).toInstant
+      }
+      override def updateValue(v: Instant, r: ResultSet, idx: Int) : Unit = {
+        r.updateTimestamp(idx, Timestamp.from(v))
+      }
+      override def valueToSQLLiteral(value: Instant) = {
+        value match {
+          case null => "NULL"
+          case _ =>  s"'${Timestamp.from(value)}'"
+        }
+      }
+    }
+
+    class OffsetTimeJdbcType extends DriverJdbcType[OffsetTime] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = {
+        /**
+          * Stores the [[OffsetTime]] as a 'VARCHAR' on databases, like SQLite, with no
+          * specific 'TIME' implementations. An example persisted value would be '18:22:43.417+01:00'.
+          * This is done because the data type 'TIME WITH TIMEZONE' is not available on all databases.
+          */
+        java.sql.Types.VARCHAR
+      }
```

Comment:
If `TIME WITH TIMEZONE` is the default, maybe it should be what we put here and override it in the drivers of the dbs that do not support it rather than the other way around. That way drivers that support it do the right thing, but forgetting to override would crash not-supporting drivers. Rather than doing it the other way around, where forgetting to override it would just do something wrong unless we have a test that finds it